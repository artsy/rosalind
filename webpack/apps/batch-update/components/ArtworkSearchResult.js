import React from 'react'
import missingImage from 'file-loader!./missing_image.png'

class ArtworkSearchResult extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (e) {
    const { artwork, onPreviewArtwork, onToggleArtwork } = this.props
    if (e.metaKey) {
      onPreviewArtwork(artwork)
    } else {
      onToggleArtwork(artwork)
    }
  }

  render () {
    const { className, artwork } = this.props
    const { name, image_url: imageUrl } = artwork
    return (
      <div className={className} onClick={this.handleClick}>
        <img src={imageUrl || missingImage} alt={name} />
        <figcaption>{name}</figcaption>
      </div>
    )
  }
}

ArtworkSearchResult.propTypes = {
  artwork: React.PropTypes.object.isRequired,
  onPreviewArtwork: React.PropTypes.func
}

/* default styled component */

import styled, { css } from 'styled-components'
import { colors } from './Layout'

const StyledArtworkSearchResult = styled(ArtworkSearchResult)`
  min-width: 115px;
  max-width: calc(20% - 1.5em);
  margin: 0.75em;
  flex: 1 1 calc(20% - 1.5em);

  img {
    width: 100%;
  }

  ${props => props.selected && css`
    background: ${colors.purple};
    outline: solid 5px ${colors.purple};
    color: white;
  `}

  figcaption {
    position: relative;
    font-size: 80%;
    max-height: 3.75em;
    line-height: 1.25em;
    overflow: hidden;
  }
`

export default StyledArtworkSearchResult
