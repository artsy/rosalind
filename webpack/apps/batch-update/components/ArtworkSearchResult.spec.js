import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import ArtworkSearchResult from './ArtworkSearchResult'

let props

beforeEach(() => {
  props = {
    artwork: {
      id: 'can',
      name: 'Soup Can',
      image_url: 'can.jpg',
      deleted: false,
      published: true,
      genomed: true
    },
    onPreviewArtwork: jest.fn(),
    onToggleArtwork: jest.fn()
  }
})

it('requires an artwork', () => {
  console.error = jest.fn()
  expect(() => {
    renderer.create(<ArtworkSearchResult />)
  }).toThrow()
  expect(console.error.mock.calls.length).toEqual(1)
})

it('renders correctly', () => {
  const rendered = renderer.create(<ArtworkSearchResult {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders in the selected state', () => {
  const rendered = renderer.create(<ArtworkSearchResult {...props} selected />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('fires the toggle handler on click', () => {
  const wrapper = mount(<ArtworkSearchResult {...props} />)
  wrapper.find('div').simulate('click')
  expect(props.onToggleArtwork.mock.calls.length).toEqual(1)
})

it('fires the preview handler on command-click', () => {
  const wrapper = mount(<ArtworkSearchResult {...props} />)
  const mockClickEvent = { metaKey: true }
  wrapper.find('div').simulate('click', mockClickEvent)
  expect(props.onPreviewArtwork.mock.calls.length).toEqual(1)
})
