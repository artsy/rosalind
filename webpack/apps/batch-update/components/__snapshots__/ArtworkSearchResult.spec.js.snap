exports[`test renders correctly 1`] = `
<div
  className="bkfdwK"
  onClick={[Function]}>
  <img
    alt="Soup Can"
    src="can.jpg" />
  <figcaption>
    Soup Can
  </figcaption>
</div>
`;

exports[`test renders in the selected state 1`] = `
<div
  className="CbrRL"
  onClick={[Function]}>
  <img
    alt="Soup Can"
    src="can.jpg" />
  <figcaption>
    Soup Can
  </figcaption>
</div>
`;
